lua/typescript-tools/init.lua

```diff
@@ -48,16 +48,14 @@ function M.setup(config)
   -- I prefer it here than in huge file tsserver.lua
   -- Rationale: `on_attach` is called based on response from `configure` request and because we
   -- have two servers nvim get also two responses
-  local on_attach_called = false
+  local last_bufnr = -1
```

Comment:
Would it be better to keep a map of `buffer number -> boolean` rather than just the last bufnr?

Reply:
Yeah, realised that closing and reopening the same buffer will result in `on_attach` not being called the second time. Will try to implement the map! One issue I see is tracking when buffers close and updating the map accordingly.

Reply:
I don't *think* you need to worry about tracking buffer closes, since buffer numbers should never be re-used.

Reply:
I have done some testing, and so far what I have found is that the buffer numbers are reused, so when reopening the same file, `on_attach` won't be called because the boolean value is still false. I have tried using an autocommand on `BufDelete` and it works, but I think it would take a lot of refactoring to move the autocommand to where the other autocommands are registered as we rely on `buf_map`.